cmd/podresdump/main.go

```diff
@@ -87,13 +75,83 @@ func main() {
 	switch *endpoint {
 	case "list":
 		Listing(dm)
-	case "watch":
-		Watching(dm)
+	case "getallocatableresources":
+		GetAllocatableResources(dm)
+	// TODO: Add watch support
+	// case "watch":
+	// 	Watching(dm)
 	default:
 		log.Fatalf("unsupported endpoint: %q", *endpoint)
 	}
 }
 
+func Listing(dm dumper) {
+	resp, err := dm.cli.List(context.TODO(), &podresourcesapi.ListPodResourcesRequest{})
+	for {
+		if err == nil {
+			break
+		} else {
+			if !dm.autoReconnect {
+				log.Fatalf("failed to list: %v", err)
+			} else {
+				log.Printf("Can't receive response: %v.Get(_) = _, %v", dm.cli, err)
+				time.Sleep(1 * time.Second)
+			}
+		}
+	}
+
+	for _, podResource := range resp.GetPodResources() {
+		if podResource.GetNamespace() != dm.namespace {
+			log.Printf("SKIP pod %q\n", podResource.Name)
+			continue
+		}
+
+		jsonBytes, err := json.Marshal(podResource)
+		if err != nil {
+			log.Printf("%v", err)
+		} else {
+			dm.out.Printf("%s\n", string(jsonBytes))
+		}
+	}
+}
+
+func GetAllocatableResources(dm dumper) {
+	resp, err := dm.cli.GetAllocatableResources(context.TODO(), &podresourcesapi.AllocatableResourcesRequest{})
+	for {
+		if err == nil {
+			break
+		} else {
+			if !dm.autoReconnect {
+				log.Fatalf("failed to show GetAllocatableResources: %v", err)
+			} else {
+				log.Printf("Can't receive response: %v.Get(_) = _, %v", dm.cli, err)
+				time.Sleep(1 * time.Second)
+			}
+		}
+	}
+
+	jsonBytes, err := json.Marshal(resp)
+	if err != nil {
+		log.Printf("%v", err)
+	} else {
+		dm.out.Printf("%s\n", string(jsonBytes))
+	}
+
+}
+
+// TODO: Add Watch Support
```

Comment:
@fromanirh I am inclining towards removing the code pertaining to watch rather than commenting it out like I have it currently. Once we finalize the design for watch support in podresource API we will have to update this anyway and it is better to leave the repo in a cleaner state.
Let me know what you think and I will leave it/make changes accordingly.

Reply:
Makes sense, thanks. No point in keeping half-baked experiments around.